lib/chef/node_map.rb

```diff
@@ -78,6 +78,12 @@ def get(node, key)
       nil
     end
 
+    def clear
```

Comment:
I usually think of this as `#reset!` when acting on singleton-ish objects like this, but `#clear` makes more sense from the hash-like interface perspective.  Could we maybe alias `:reset!` to this?